tests/handlers/external_test.py

```diff
@@ -0,0 +1,244 @@
+
+    r = await client.get(f"/moneypenny/{dossier.username}")
+    assert r.status_code == 202
+
+    assert_kubernetes_objects_are(
+        mock_kubernetes,
+        "ConfigMap",
+        [
+            V1ConfigMap(
+                metadata=V1ObjectMeta(
+                    name=f"{dossier.username}-cm", namespace="default"
+                ),
+                data={
+                    "dossier.json": json.dumps(
+                        dossier.dict(), sort_keys=True, indent=4
+                    )
+                },
+            )
+        ],
+    )
+    assert_kubernetes_objects_are(
+        mock_kubernetes,
+        "Pod",
+        [
+            V1Pod(
+                metadata=V1ObjectMeta(
+                    name=f"{dossier.username}-pod",
+                    namespace="default",
+                ),
+                spec=V1PodSpec(
+                    automount_service_account_token=False,
+                    containers=[
+                        {
+                            "name": "farthing",
+                            "image": "lsstsqre/farthing",
+                            "securityContext": {
+                                "runAsUser": 1000,
+                                "runAsNonRootUser": True,
+                                "allowPrivilegeEscalation": False,
+                            },
+                            "volumeMounts": [
+                                {
+                                    "mountPath": "/opt/dossier",
+                                    "name": f"dossier-{dossier.username}-vol",
+                                    "readOnly": True,
+                                }
+                            ],
+                        }
+                    ],
+                    image_pull_secrets=[],
+                    init_containers=[],
+                    restart_policy="OnFailure",
+                    security_context=V1PodSecurityContext(
+                        run_as_group=1000, run_as_user=1000
+                    ),
+                    volumes=[
+                        V1Volume(
+                            name=f"dossier-{dossier.username}-vol",
+                            config_map=V1ConfigMapVolumeSource(
+                                default_mode=0o644,
+                                name=f"{dossier.username}-cm",
+                            ),
+                        )
+                    ],
+                ),
+                status=V1PodStatus(phase="Running"),
+            )
+        ],
+    )
+
+    r = await client.get(f"/moneypenny/{dossier.username}")
+    assert r.status_code == 202
+
+    pod_name = f"{dossier.username}-pod"
+    pod = await mock_kubernetes.read_namespaced_pod(pod_name, "default")
+    pod.status = V1PodStatus(phase="Succeeded")
+
+    r = await client.get(f"/moneypenny/{dossier.username}")
+    assert r.status_code in (200, 404)
+
+    timeout = datetime.now(tz=timezone.utc) + timedelta(seconds=5)
```

Comment:
It's just to give the background thread time to run (which will normally be way faster than 5s; I just usually use 5s as the window around anything time-related in tests since it seems to be robust against overloaded test runners). I'll add a comment.